datalab/datalab.sh

```diff
@@ -28,53 +38,65 @@ function update_apt_get() {
   return 1
 }
 
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-PROJECT=$(/usr/share/google/get_metadata_value ../project/project-id)
-DOCKER_IMAGE=$(/usr/share/google/get_metadata_value attributes/docker-image || true)
-SPARK_PACKAGES=$(/usr/share/google/get_metadata_value attributes/spark-packages || true)
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
 
 if [[ "${ROLE}" == 'Master' ]]; then
-  update_apt_get
+  update_apt_get || err 'Failed to update apt-get'
+  mkdir -p "${DATALAB_DIR}"
+
   apt-get install -y -q docker.io
+  if [ $? != 0 ]; then
```

Comment:
Your convention seems to be `<command> || err 'Reason'`, which is a little bit shorter